### ISSUE TITLE
Numerical Recipes

### ISSUE THREAD

**Issue by leonfoks** [fortran-lang/stdlib#115]
I wanted to bring this up for those that may not be familiar with the licensing for Numerical Recipes algorithms. See this http://numerical.recipes/licenses/

Do not use them for anything stdlib.  The license for all those algorithms is only for single use or institutional use basis, re-distributing them is not allowed.

I wonder how much money they make.  They should just move to an open source license...
Perhaps this brings up the sticky subject of licensing? But I don't want to start it.

**Comment by certik** [comment]
Indeed, we cannot use any code from Numerical Recipes.

All stdlib must be MIT/BSD style licensed. No GPL, no LGPL, no other more restrictive licenses.

The idea would be that vendors could provide commercial versions of stdlib with faster implementations.

**Comment by leonfoks** [comment]
Using public domain code (or whatever public domain means, it's not consistent for different countries!)
From what I gather, we can include "PD" code into the stdlib source.  But we cannot re-license that code since we are not the original authors.  I think we would have to maintain the public domain attribution to those pieces of code and state that in those files?

**Comment by milancurcic** [comment]
https://en.m.wikipedia.org/wiki/Public-domain_software

I think this means we can do whatever we want with PD code, including relicensing.

**Comment by leonfoks** [comment]
okay great!

**Comment by nncarlson** [comment]
Back around 2015 I was involved in some contract talks with a major corporation, and their IP lawyer shared a [couple slides](https://user-images.githubusercontent.com/7231327/72691041-56681080-3adf-11ea-8e48-293320560df8.png) (redacted) with me that described industry's view of open source software and its various licenses. You might find it interesting. Bottom line MIT and BSD are good, unsurprisingly.

**Comment by masuday** [comment]
How and where can I put the license information (the copyright and the original authors) when the code is taken from *open source* libraries? A separate question is whether I clarify the original author of a public-domain program.

Perhaps we can have a file (`SOURCE.md`) to have the copyright notice instead of putting it to the source code. We can optionally put the source of public-domain software to the file just for reference.

**Comment by dmey** [comment]
> How and where can I put the license information (the copyright and the original authors) when the code is taken from open source libraries? A separate question is whether I clarify the original author of a public-domain program.

It depends -- this is normally the case only if you bundle external code.

> Perhaps we can have a file (SOURCE.md) to have the copyright notice instead of putting it to the source code. We can optionally put the source of public-domain software to the file just for reference.

General good practice is to have something like https://github.com/microsoft/vscode/blob/master/ThirdPartyNotices.txt.

**Comment by masuday** [comment]
I agree it depends. I meant the BSD-like license. I believe that I should write the copyright notice somewhere in the package when I have (wholly or partially) taken or modified the original code released under the BSD license. I was looking for the good practice that you pointed.

**Comment by certik** [comment]
I've been putting this into the LICENSE file directly: https://github.com/certik/hfsolver/blob/b4c50c1979fb7e468b1852b144ba756f5a51788d/LICENSE, but I like the vscode's approach also: https://github.com/microsoft/vscode/blob/a14ad2f8b0a83bace3dae6c83c49224a3fb870aa/ThirdPartyNotices.txt as @dmey suggested.

**Comment by nncarlson** [comment]
I think it's also very important to maintain documentation of the author and license (though not necessarily the full text but simply a reference, as in "BSD") in the individual source files as well.
